Assignment4/Assignment4/tests/CourseTest.cs

```diff
@@ -0,0 +1,124 @@
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+
+namespace tests
+{
+    [TestClass]
+    public class TestCourse
+    {        
+        private static Assignment4.Course Course { get; set; }
+
+        [TestInitialize]
+        public void SetupUnivCourse()
+        {
+            Course = new Assignment4.Course("Computer Science", ".net", 208, 21, 2, 4);
+            Course.resetCourseTotal();
+        }
+
+        [TestMethod]
+        public void TestValid_SetSubject()
+        {
+            Course.Subject = "Chemistry";
+            Assert.AreEqual("Chemistry", Course.Subject);
+        }
+
+        [TestMethod]
+        [ExpectedException(typeof(NotSupportedException))]
+        public void TestInvalid_SetSubjectEmpty()
+        {
+            Course.Subject = "";
+            Assert.Fail();
+        }
+
+        [TestMethod]
+        [ExpectedException(typeof(NullReferenceException))]
+        public void TestInvalid_SetSubjectNull()
+        {
+            Course.Subject = null;
+            Assert.Fail();
+        }
+ 
+        [TestMethod]
+        public void TestValid_ReadNumTeacher()
+        {
+            int actual = Course.NumTeachers;
+            Assert.AreEqual(1, actual);
+        }
+ 
+        [TestMethod]
+        public void TestValid_NumStudentGet()
+        {
+            int actual = Course.NumStudents;
+            Assert.AreEqual(21, actual);
+        }
+
+        [TestMethod]
+        public void TestValid_NumStudentSet()
+        {
+            Course.NumStudents = 42;
+            Assert.AreEqual(42, Course.NumStudents);
+        }
+
+        [TestMethod]
+        public void TestValid_GetRoomNum()
+        {
+            int actual = Course.RoomNum;
+            Assert.AreEqual(208, actual);
+        }
+
+        [TestMethod]
+        [ExpectedException(typeof(ArgumentOutOfRangeException))]
+        public void TestInvalid_SetRoomNum()
+        {
+            Course.RoomNum = 42;
+            Assert.Fail();
+        }
+
+        [TestMethod]
+        public void TestValid_SetRoomNum()
+        {
+            Course.RoomNum = 123;
+            int actual = Course.RoomNum;
```

Comment:
You dont really need to store the RoomNum into a temporary variable.
```suggestion
             Assert.AreEqual(123, Course.RoomNum)
```